### PR TITLE
Analysis Plan  Feedback

### PR DESCRIPTION
**Pull request by kescobo**
:exclamation: Don't merge this! :exclamation: 

This PR is for feeback related to your analysis plan

## Overall

- [x] 2 analysis plans
- [ ] `analysis_plan.md` file, turned in on time

## Plan 1

- [x]  What analysis you plan to do
- [x] What functions you will use, including any new functions that you need to write
- [x] How you intend to display the results (eg what will you plot, or how will you summarize the results)

## Plan 2

- [x]  What analysis you plan to do
- [x] What functions you will use, including any new functions that you need to write
- [x] How you intend to display the results (eg what will you plot, or how will you summarize the results)

#### Excellent - full credit (5 pts)

- [x] Two proposed analyses are clear and concise,
  and contain details including descriptions of functions needed,
  plots or other display items intended,
  and a clear roadmap for how to generate them
- [x] Markdown formatting is used effectively to organize into subsections,
  emphasize important points,
  and provide concise descriptions in lists
- [x] Assignment is submitted on time

#### Good - partial credit (4 pts)

- [ ] Two proposed analyses are clear and concise, but are missing some details
  of functions needed, plots or other display items intended
- [ ] Roadmap for generating intended analyses is clear,
  but missing some steps, or is complete but not entirely clear.
- [ ] Markdown formatting is used, but not as effectively as possible
- [ ] Assignment is submitted on time

#### Needs work - partial credit (3 or fewer points)

- [ ] Only 1 analysis proposed, or descriptions of analyses are incomplete
- [ ] Descriptions of functions needed and plots or other display items
  are unclear or incomplete
- [ ] Markdown formatting is used sparingly or not at all.
- [ ] Assignment is submitted late, or incorrectly.
z

### CODE REVIEW

File: final-project-analysis-proposals.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        - Method(s) for data visualization
+    2. Subquestion of analysis 2
+        - A list of Julia function(s) utilized
+        - Method(s) for data visualization
+- Explanation of Julia function(s) to be written (if applicable) 
+    - Parameters/compatible inputs
+    - Output returned by function
+- Miscellaneous notes (if applicable)
+
+#
```
Comment (kescobo) [inline final-project-analysis-proposals.md:22]: Don't think we want this hash (or the one on ln 24)

File: final-project-analysis-proposals.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+[1]: https://cen.acs.org/biological-chemistry/biochemistry/Researchers-in-China-report-structure-of-the-novel-coronavirus-bound-to-its-human-target/98/web/2020/03
+Because a protein’s amino acid sequencing determines its tertiary/quaternary structures which consequently determines its interactions with other proteins, this implies that the SARS-CoV2 and SARS-CoV spike proteins differ considerably in their sequencing. However, because these two viruses target the same host protein, ACE2, we also know that their sequences will not drastically different. I am curious to find out the degree of similarity between the spike protein sequences of these two viruses.
+
+### **Sub-questions of Analysis**
```
Comment (kescobo) [inline final-project-analysis-proposals.md:31]: No need to bold this, the header takes care of the formattng

File: final-project-analysis-proposals.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+**1. How well do these two protein sequences align?**
+- **Julia function(s) utilized:** 
+    - Needleman-Wunsch or Smith-Waterman alignment algorithm
+        - Inputs: SARS-CoV2 spike protein sequence, SARS-CoV spike protein sequence
```
Comment (kescobo) [inline final-project-analysis-proposals.md:35]: How are you going to pull the spike out of the genomes and translate them into protein? Or do you have a different source for spike protein sequences?

File: final-project-analysis-proposals.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+- **Method(s) for data visualization:**
+    - a simplified version of a **[bioinformatics dot-plot][2]**, where the SARS-CoV2 spike protein sequence lies on the x-axis and the SARS-CoV spike protein sequence lies on the y-axis with the extreme N-terminal residues near the origin, and a dot is plotted for every position with matching residues; as a result, consecutively matching residues (i.e. matching subsequences) will be displayed as diagonal lines pointing from bottom left corner to upper right
```
Comment (kescobo) [inline final-project-analysis-proposals.md:37]: I've seen these before, and they're very cool! I don't know if there are any packages for creating them though. Do you have a sense of how to convert alignments into numberical data to do this? Might be able to make use of a `heatmap` plot, and a grid of 1s and 0s, maybe

File: final-project-analysis-proposals.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+- `mis_matchSubsequences(alignedSeq)`
+    - Compatible Input: takes a tuple of 2 aligned sequences, where each sequence is of type `String`
+    - Compatible Output: a dictionary with an indeterminate number of key-value pairs where 
+        - each value is an `Array` of **matching/mismatching protein subsequence(s)** as `Strings`
+        - each key is a **3-element** `Tuple` with **1)** whether it is a “Match” or “Mismatch”, **2)** the corresponding position(s) of the subsequence as a string, and **3)** the length of this subsequence as an `Int`
+    
+    > Note 1: If the residue subsequences are a match, the value Array will contain 1 String, if they are a mismatch, the Array will contain 2 Strings
+
+    > Note 2: for example, given an `alignedSeq` of (MES`LV`PGFNEKTH`V`QLS, MES`--`PGFNEKTH`A`QLS), `mis_matchSubsequences(alignedSeq)` returns {(“Match”, “1-3”, 3)=>[“MES”]; (“Mismatch”, “4-5”, 2)=>[“LV”, “--”]; (“Match”, “6-13”, 8)=>[“PGFNEKTH”]; (“Mismatch”, “14”, 1)=>[“V”, “A”]; (“Match”, “15-17”, 3)=>[“QLS”]}
```
Comment (kescobo) [inline final-project-analysis-proposals.md:69]: This seems kinda like [CIGAR notation](https://genome.sph.umich.edu/wiki/SAM#What_is_a_CIGAR.3F)... 

Seems like a lot of data to keep track of. Also, the keys seem like a lot... how are you going to go back in and access it? Especially keeping in mind that dictionary keys are not ordered.

You might consider doing something like

```
struct AlignmentSegment
    type::String # one of "Match", "Mismatch"
    positions::Range #eg 6:13
    sequences::Vector
end
```
and then store those in an array or something

File: final-project-analysis-proposals.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        - Input (x 50): SARS-CoV2 genomic sequence from State A, B, C, D, E, ...
+        - Input: the first SARS-CoV2 genomic sequence ever collected in Seattle, WA on January 21, 2020 
+    - `Kmer set distance function`
+        > will need to run this function 50 times to generate 50 distance metrics between each State sample and the Seattle, WA sample
+        - Inputs: kmer sets from the SARS-CoV2 genome collected from a single sample in State A and the first sample ever collected in Seattle, WA on January 21, 2020 
+- **Method(s) for data visualization:**
+    - A **statistical dot plot** of **50 data pairs**, where each data pair consists of 
+        1. the distance between the State of sample collection and Seattle, WA plotted on the Y-axis
+        2. the distance metric of these SARS-CoV2 genomes plotted on the X-axis
+    - A **linear regression** will be run with these **50 data pairs**, and the **line of best fit** resulting from this linear regression will be plotted on the statistical dot plot mentioned above
```
Comment (kescobo) [inline final-project-analysis-proposals.md:90]: see `GLM.jl`

File: final-project-analysis-proposals.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    
+    > Note 1: If the residue subsequences are a match, the value Array will contain 1 String, if they are a mismatch, the Array will contain 2 Strings
+
+    > Note 2: for example, given an `alignedSeq` of (MES`LV`PGFNEKTH`V`QLS, MES`--`PGFNEKTH`A`QLS), `mis_matchSubsequences(alignedSeq)` returns {(“Match”, “1-3”, 3)=>[“MES”]; (“Mismatch”, “4-5”, 2)=>[“LV”, “--”]; (“Match”, “6-13”, 8)=>[“PGFNEKTH”]; (“Mismatch”, “14”, 1)=>[“V”, “A”]; (“Match”, “15-17”, 3)=>[“QLS”]}
+
+#
+## Q2: Is there a correlation between the distance between the location of where a SARS-CoV2 sample was collected and the location of the earliest reported COVID case in the U.S. (Seattle, WA) and the number of genomic residue mismatches? If so, is the correlation positive or negative? (i.e. positive = the farther away from Seattle, WA a sample is collected, the more different the SARS-CoV2 genomes are)
+#
+> Note 1: NCBI does not label samples with the exact location (e.g. City, State) of where they were collected, so the distance portion of this analysis will be the distance in miles between Seattle, WA, the location of the earliest reported COVID case in the U.S., and State A as displayed on Google maps
```
Comment (kescobo) [inline final-project-analysis-proposals.md:74]: Are you going to get these by hand?

**Comment by kescobo** [comment]
This is a pretty ambitious plan, I suspect you'll need to walk it back a bit to actually accomplish anything. Be sure to keep track of your work in notebooks, including the spots where you struggle or run into problems.

File: final-project-analysis-proposals.md (hunk 5 above)
Comment (lushaojia) [reply final-project-analysis-proposals.md:69]: I totally did not even think about how to go back in and access the data when I first came up with this function; will look into doing it with a struct (though I am pretty inexperienced with them)!

File: final-project-analysis-proposals.md (hunk 7 above)
Comment (lushaojia) [reply final-project-analysis-proposals.md:74]: I think so!

File: final-project-analysis-proposals.md (hunk 3 above)
Comment (lushaojia) [reply final-project-analysis-proposals.md:35]: I believe on NCBI there is a separate tab for data on protein sequences! For SARS-CoV2, the spike protein is called "surface glycoprotein" (I see one or two on the first page); for SARS-CoV, I believe it's just called "spike protein"!

Comment (kescobo) [reply final-project-analysis-proposals.md:35]: Sounds good!

File: final-project-analysis-proposals.md (hunk 5 above)
Comment (kescobo) [reply final-project-analysis-proposals.md:69]: What an opportunity to learn! 

You could also just use `BioAlignments.jl` and use their representation - up to you

**Pull request closed without merging**